bundles/org.eclipse.jface/src/org/eclipse/jface/resource/ResourceManager.java

```diff
@@ -102,51 +102,172 @@ public abstract class ResourceManager {
+	 * This method should only be used for resources that should remain allocated
+	 * for the lifetime of this resource-manager. To allocate shorter-lived
 	 * resources, manage them with <code>create</code>, and <code>destroy</code>
 	 * rather than this method.
 	 * </p>
 	 *
 	 * <p>
-	 * This method should never be called on the global resource manager,
-	 * since all resources will remain allocated for the lifetime of the app and
-	 * will be effectively leaked.
+	 * This method should never be called on the global resource-manager, since all
+	 * resources will remain allocated for the lifetime of the app and will be
+	 * effectively leaked.
 	 * </p>
 	 *
 	 * @param descriptor identifier for the requested resource
-	 * @return the requested resource. Never null.
+	 * @return the requested resource, never null
 	 * @throws DeviceResourceException if the resource does not exist yet and cannot
-	 * be created for any reason.
+	 *                                 be created for any reason.
 	 *
 	 * @since 3.3
+	 * @see ResourceManager#get(ImageDescriptor)
+	 * @see ResourceManager#get(ColorDescriptor)
+	 * @see ResourceManager#get(FontDescriptor)
 	 */
 	public final Object get(DeviceResourceDescriptor descriptor) {
-		Object result = find(descriptor);
+		return getCachedResource(descriptor);
+	}
 
-		if (result == null) {
-			result = create(descriptor);
-		}
+	/**
+	 * Returns a previously-allocated {@link Image} or allocates a new one if none
+	 * exists yet. The image will remain allocated for at least the lifetime of this
+	 * resource-manager. If necessary, the image will be deallocated automatically
+	 * when this resource-manager is disposed.
+	 * <p>
+	 * The images returned from this method are reference counted and may be shared
+	 * internally with other resource-managers. They should never be disposed
+	 * outside of the ResourceManager framework, or it will cause exceptions in
+	 * other code that shares them. For example, never call
+	 * {@link org.eclipse.swt.graphics.Image#dispose()} on anything returned from
+	 * this method.
+	 * </p>
+	 *
+	 * <p>
+	 * This method should only be used for images that should remain allocated for
+	 * the lifetime of this resource-manager. To allocate shorter-lived images,
+	 * manage them with {@link #createImage(ImageDescriptor)}, and
+	 * {@link #destroyImage(ImageDescriptor)} rather than this method.
+	 * </p>
+	 *
+	 * <p>
+	 * This method should never be called on the global resource-manager, since all
+	 * resources will remain allocated for the lifetime of the app and will be
+	 * effectively leaked.
+	 * </p>
+	 *
+	 * @param descriptor identifier for the requested images
+	 * @return the requested resource, never null.
+	 * @throws DeviceResourceException if the images does not exist yet and cannot
+	 *                                 be created for any reason.
+	 *
+	 * @since 3.31
+	 */
+	public final Image get(ImageDescriptor descriptor) {
```

Comment:
This method should get another name then get(DeviceResourceDescriptor) because ImageDescriptor extends ImageDescriptor. A call that already used get(DeviceResourceDescriptor) with an ImageDescriptor would otherwise now get another (narrower) result which can result in an error message like unnecessary typecast.
```suggestion
	public final Image getImage(ImageDescriptor descriptor) {
```